pkg/util/kubernetes/apiserver/apiserver.go

```diff
@@ -674,3 +690,181 @@ func (c *APIClient) GetARandomNodeName(ctx context.Context) (string, error) {
+	var b bytes.Buffer
+	var userConfig chartUserValues
+	var selector labels.Selector
+	var helmConfigmap *v1.ConfigMap
+
+	// Only a single release for a given name can be deployed at one time.
+	// Other releases present a different status.
+	selector = labels.Set{
+		"owner":  "helm",
+		"status": "deployed",
+		"name":   name,
+	}.AsSelector()
+
+	configmaps, err := getConfigmapList(cl, namespace)
+	if err != nil {
+		log.Errorf("Can't list configmaps from the API server: %s", err.Error())
+	}
+
+	for _, cm := range configmaps {
+		if selector.Matches(labels.Set(cm.ObjectMeta.Labels)) {
+			helmConfigmap = &cm
+			// Chart data is stored in a gzipped base 64 encoded string
+			b64EncodedGzipRelease := helmConfigmap.Data["release"]
+			decodedrelease := decodeRelease(b64EncodedGzipRelease)
+			err := json.Unmarshal(decodedrelease, &userConfig)
+			if err != nil {
+				log.Debugf("Unable to retrieve the config data: %s", err.Error())
+			}
+			configjson, err := json.Marshal(userConfig)
+			if err != nil {
+				log.Debugf("Can't marshall user values into a proper JSON: %s", err.Error())
+			}
+			configyaml, err := yaml.JSONToYAML(configjson)
+			if err != nil {
+				log.Debugf("Can't convert user values to YAML: %s", err.Error())
+			}
+			fmt.Fprintln(&b, string(configyaml))
+			return b.Bytes(), nil
+		}
+	}
+
+	// If we exit the loop and reach this point, a matching configmap couldn't be found
+	return nil, log.Error("No matching configmap found")
+}
+
+// Retrieve Helm chart user values from the API server for a given name and namespace looking through secrets
+func GetDatadogHelmUserValuesFromSecret(cl *APIClient, name string, namespace string) ([]byte, error) {
+	var b bytes.Buffer
+	var userConfig chartUserValues
+	var selector labels.Selector
+	var helmSecret *v1.Secret
+
+	// Only a single release for a given name can be deployed at one time.
+	// Other releases present a different status.
+	selector = labels.Set{
+		"owner":  "helm",
+		"status": "deployed",
+		"name":   name,
+	}.AsSelector()
+
+	secrets, err := getSecretList(cl, namespace)
+	if err != nil {
+		log.Errorf("Can't list secrets from the API server: %s", err.Error())
+	}
+
+	for _, secret := range secrets {
+		if selector.Matches(labels.Set(secret.ObjectMeta.Labels)) {
+			helmSecret = &secret
+			// Contrary to the configmap, the chart data is encoded twice in base64 (native Kubernetes secret base64 + Helm).
+			// The K8s Go client decodes the native Kubernetes encoding, so we do not have to do it ourselves and can use the same logic, as the ConfigMap
+			b64EncodedGzipRelease := helmSecret.Data["release"]
+			decodedrelease := decodeRelease(string(b64EncodedGzipRelease))
+			err = json.Unmarshal(decodedrelease, &userConfig)
+			if err != nil {
+				log.Debugf("Unable to retrieve the config data: %s", err.Error())
+			}
+			configjson, err := json.Marshal(userConfig)
+			if err != nil {
```

Comment:
same here, and for the line 861 if, you have an error you should just `continue`